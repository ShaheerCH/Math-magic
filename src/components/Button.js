import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = null;
  }

  render() {
    const { bgColor, btnRef, handleClick } = this.props;
    return (
      <button type="button" className={bgColor} onClick={handleClick}>
        {btnRef}
      </button>
    );
  }
}

Button.propTypes = {
  bgColor: PropTypes.string.isRequired,
  btnRef: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
