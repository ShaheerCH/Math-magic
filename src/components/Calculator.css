.calculator {
  display: inline-grid;
  grid-template-columns: repeat(4, 60px);
  grid-template-rows: repeat(6, 60px);
  gap: 1px;
  background-color: #c2c2c2;
}

.calculator > * {
  border: none;
}

.screen {
  grid-column: 1 / span 4;
  background-color: #858694;
  font-size: 1.5rem;
  color: white;
  padding: 20px 8px;
  text-align: right;
}

.zero {
  grid-column: 1 / span 2;
}

.btn-orange {
  background-color: #f5913e;
}

.btn-gray {
  background-color: #e0e0e0;
}
