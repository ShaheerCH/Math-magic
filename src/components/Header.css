h1 {
  width: 23vw;
  margin-right: 46vw;
  font-size: 1.9rem;
}

header {
  width: 82vw;
  margin: 0 auto;
  display: flex;
  align-items: center;
}

nav {
  width: 25vw;
  display: flex;
  justify-content: space-between;
  /* stylelint-disable-next-line no-missing-end-of-source-newline */
}