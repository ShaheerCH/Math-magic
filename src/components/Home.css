.home {
  width: 90vw;
  margin: 0 auto;
}
