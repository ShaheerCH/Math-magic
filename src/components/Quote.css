.quote p {
  width: 60vw;
  font-size: 1.7rem;
}

.quote {
  display: flex;
  justify-content: center;
}
