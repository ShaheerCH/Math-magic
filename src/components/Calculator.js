import './Calculator.css';
import React, { useState } from 'react';
import Button from './Button';
import Screen from './Screen';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: 0,
    operation: null,
  });

  const onButtonPress = ({ target }) => {
    setState((state) => calculate(state, target.innerText));
  };

  const { total, next, operation } = state;

  return (
    <article className="calculator">
      <Screen total={total} next={next} operation={operation} />
      <Button bgColor="btn-gray" btnRef="AC" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="+/-" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="%" handleClick={onButtonPress} />
      <Button bgColor="btn-orange" btnRef="รท" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="7" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="8" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="9" handleClick={onButtonPress} />
      <Button bgColor="btn-orange" btnRef="x" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="4" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="5" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="6" handleClick={onButtonPress} />
      <Button bgColor="btn-orange" btnRef="-" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="1" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="2" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="3" handleClick={onButtonPress} />
      <Button bgColor="btn-orange" btnRef="+" handleClick={onButtonPress} />
      <Button bgColor="btn-gray zero" btnRef="0" handleClick={onButtonPress} />
      <Button bgColor="btn-gray" btnRef="." handleClick={onButtonPress} />
      <Button bgColor="btn-orange" btnRef="=" handleClick={onButtonPress} />
    </article>
  );
};

export default Calculator;
