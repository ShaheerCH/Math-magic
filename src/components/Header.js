import './Header.css';
import React from 'react';
import { Link } from 'react-router-dom';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = null;
  }

  render() {
    return (
      <header>
        <h1>Math Magicians</h1>
        <nav>
          <Link to="/">Home</Link>
          <Link to="/Calculator">Calculator</Link>
          <Link to="/Quote">Quote</Link>
        </nav>
      </header>
    );
  }
}

export default Header;
