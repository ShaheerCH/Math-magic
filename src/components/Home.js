import './Home.css';
import React from 'react';
import Header from './Header';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = null;
  }

  render() {
    return (
      <>
        <Header />
        <main className="home">
          <h2>Welcome to our page!</h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            tincidunt quis elit nec finibus. Pellentesque habitant morbi
            tristique senectus et netus et malesuada fames ac turpis egestas.
            Aenean cursus elit eget sem eleifend scelerisque nec et ex. Morbi
            non odio consequat, pellentesque neque vel, finibus ipsum. Maecenas
            vestibulum nisl tellus, sed scelerisque turpis tempor eu. Donec
            tempor nibh eget ligula accumsan pellentesque. Curabitur id
            convallis nulla. Praesent lectus tellus, vulputate eget blandit
            eget, consequat sit amet velit. Curabitur lectus magna, rutrum eu
            ante quis, ullamcorper vehicula nisi.
          </p>
          <p>
            Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis
            et minim exercitation in incididunt amet esse aute. Sunt nostrud
            reprehenderit est reprehenderit commodo. Aute cillum aliquip enim
            occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi
            in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco
            non velit ullamco deserunt sint.
          </p>
        </main>
      </>
    );
  }
}

export default Home;
