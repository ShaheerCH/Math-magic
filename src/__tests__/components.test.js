import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';
import Button from '../components/Button';
import Screen from '../components/Screen';

describe('Header renders correctly', () => {
  test('Renders Header Correctly', () => {
    const header = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(header).toMatchSnapshot();
  });
});

describe('Home renders correctly', () => {
  test('Renders Header Correctly', () => {
    const home = render(
      <Router>
        <Home />
      </Router>,
    );
    expect(home).toMatchSnapshot();
  });
});

describe('Quote renders correctly', () => {
  test('Renders Header Correctly', () => {
    const quote = render(
      <Router>
        <Quote />
      </Router>,
    );
    expect(quote).toMatchSnapshot();
  });
});

describe('Calculator renders correctly', () => {
  test('Renders Calculator Correctly', () => {
    const calc = render(
      <Router>
        <Calculator />
      </Router>,
    );
    expect(calc).toMatchSnapshot();
  });
});

describe('Button renders correctly', () => {
  test('Renders Button Correctly', () => {
    const btn = render(
      <Router>
        <Button />
      </Router>,
    );
    expect(btn).toMatchSnapshot();
  });
});

describe('Screen renders correctly', () => {
  test('Renders Screen Correctly', () => {
    const scr = render(
      <Router>
        <Screen />
      </Router>,
    );
    expect(scr).toMatchSnapshot();
  });
});
