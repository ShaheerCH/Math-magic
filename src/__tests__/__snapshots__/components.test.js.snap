// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly Renders Button Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        type="button"
      />
    </div>
  </body>,
  "container": <div>
    <button
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Calculator renders correctly Renders Calculator Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <main
        class="calc"
      >
        <h2>
          Let's do some math!
        </h2>
        <article
          class="calculator"
        >
          <div
            class="screen"
          >
            0
          </div>
          <button
            class="btn-gray"
            type="button"
          >
            AC
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            +/-
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            %
          </button>
          <button
            class="btn-orange"
            type="button"
          >
            รท
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            7
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            8
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            9
          </button>
          <button
            class="btn-orange"
            type="button"
          >
            x
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            4
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            5
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            6
          </button>
          <button
            class="btn-orange"
            type="button"
          >
            -
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            1
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            2
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            3
          </button>
          <button
            class="btn-orange"
            type="button"
          >
            +
          </button>
          <button
            class="btn-gray zero"
            type="button"
          >
            0
          </button>
          <button
            class="btn-gray"
            type="button"
          >
            .
          </button>
          <button
            class="btn-orange"
            type="button"
          >
            =
          </button>
        </article>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
    <main
      class="calc"
    >
      <h2>
        Let's do some math!
      </h2>
      <article
        class="calculator"
      >
        <div
          class="screen"
        >
          0
        </div>
        <button
          class="btn-gray"
          type="button"
        >
          AC
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          %
        </button>
        <button
          class="btn-orange"
          type="button"
        >
          รท
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          7
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          8
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          9
        </button>
        <button
          class="btn-orange"
          type="button"
        >
          x
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          4
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          5
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          6
        </button>
        <button
          class="btn-orange"
          type="button"
        >
          -
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          1
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          2
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          3
        </button>
        <button
          class="btn-orange"
          type="button"
        >
          +
        </button>
        <button
          class="btn-gray zero"
          type="button"
        >
          0
        </button>
        <button
          class="btn-gray"
          type="button"
        >
          .
        </button>
        <button
          class="btn-orange"
          type="button"
        >
          =
        </button>
      </article>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <main
        class="home"
      >
        <h2>
          Welcome to our page!
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tincidunt quis elit nec finibus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aenean cursus elit eget sem eleifend scelerisque nec et ex. Morbi non odio consequat, pellentesque neque vel, finibus ipsum. Maecenas vestibulum nisl tellus, sed scelerisque turpis tempor eu. Donec tempor nibh eget ligula accumsan pellentesque. Curabitur id convallis nulla. Praesent lectus tellus, vulputate eget blandit eget, consequat sit amet velit. Curabitur lectus magna, rutrum eu ante quis, ullamcorper vehicula nisi.
        </p>
        <p>
          Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
    <main
      class="home"
    >
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tincidunt quis elit nec finibus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aenean cursus elit eget sem eleifend scelerisque nec et ex. Morbi non odio consequat, pellentesque neque vel, finibus ipsum. Maecenas vestibulum nisl tellus, sed scelerisque turpis tempor eu. Donec tempor nibh eget ligula accumsan pellentesque. Curabitur id convallis nulla. Praesent lectus tellus, vulputate eget blandit eget, consequat sit amet velit. Curabitur lectus magna, rutrum eu ante quis, ullamcorper vehicula nisi.
      </p>
      <p>
        Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <main
        class="quote"
      >
        <blockquote>
          <p>
            Heisenberg "Whats My Name You are Goddamn right "
          </p>
        </blockquote>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
    <main
      class="quote"
    >
      <blockquote>
        <p>
          Heisenberg "Whats My Name You are Goddamn right "
        </p>
      </blockquote>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Screen renders correctly Renders Screen Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="screen"
      >
        0
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="screen"
    >
      0
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
